### PR TITLE
add preliminar support for nvme

### PR DESCRIPTION
**Pull request by mudler**
* snapshots/migration on nvme are still not supported, currently breaking
milestones tests

Proof:
partitioning

Local machine test: [Test](http://10.100.51.55/tests/23/)
Log: [autoinst-log.txt](https://github.com/os-autoinst/os-autoinst/files/992781/autoinst-log.txt)

Progress related issue: [Poo#18988](https://progress.opensuse.org/issues/18988)

### CODE REVIEW

**Comment by coolo** [comment]
Before @okury gets you, read the Gospel of Chris:  https://chris.beams.io/posts/git-commit/

**Comment by coolo** [comment]
(and get extra points if you take the opportunity to refactor this start_qemu function a bit by extracting e.g. network and storage)

**Comment by mudler** [comment]
Could have been [worse](http://whatthecommit.com/) :)
I would restart refactor this one, but in a separate commit i guess

**Comment by mudler** [comment]
Attaching [autoinst-log.txt](https://github.com/os-autoinst/os-autoinst/files/993337/autoinst-log.txt) of a full build with nvme support

File: autotest.pm [hunk 1]
```diff
@@ -214,6 +214,13 @@ sub runalltests {
     my $snapshots_supported = query_isotovideo('backend_can_handle', {function => 'snapshots'});
     bmwqemu::diag "Snapshots are " . ($snapshots_supported ? '' : 'not ') . "supported";
 
+    # XXX: Temporary (hopefully) workaround for nvme, since snapshots fails.
+    # See also https://github.com/os-autoinst/os-autoinst/pull/781
+    if ($bmwqemu::vars{HDDMODEL} eq "nvme") {
```
Comment (coolo) [inline autotest.pm]: this belongs in backend/qemu's can_handle function

**Comment by mudler** [comment]
Attaching [autoinst-log.txt](https://github.com/os-autoinst/os-autoinst/files/995364/autoinst-log.txt) of a [full-build](http://10.100.51.55/tests/26) with nvme support for the latest changes

**Pull request merged**